### DISCUSSION TITLE
Is there a way to configure user lockout settings per tenant?

### DISCUSSION THREAD

**Discussion by mariojsnunes** [OrchardCMS/OrchardCore#10229] [Q&A] [answered]
The after reading the docs, it seems to be global for all tenants

**Comment by ns8482e** [comment, marked as answer]
IMHO it could be possible - and the current documentation missed for multi tenant - may be because it's obvious to achieve by implementing Option Pattern? https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-5.0 

You can modify any `IOptions` values by implementing `IConfigureOptions` or `IPostConfigureOptions`


I have not tested  - but you can try following and see if that works for you. If implementing `IConfigureOptions`  doesn't work then try implementing `IPostConfigureOptions`

1. Define `TenantIdentity` options

```cs
internal class TenantIdentityOptions : IConfigureOptions<IdentityOptions>
{
     private readonly IShellConfiguration _shellConfiguration;
     public TenantIdentityOptions(IShellConfiguration shellConfiguration)
     {
         _shellConfiguration = shellConfiguration;
     }
     public void Configure(IdentityOptions options)
     {
           _shellConfiguration.GetSection("IdentityOptions").Bind(options);
     }
}
```

2. Configure in App's startup in tenant pipeline or it can also be configured in module startup.

```cs
    services.ConfigureOptions<TenantIdentityOptions>();
```

3. Define following configuration in `appsettings.json` for each tenant.

```json
{
  "OrchardCore": {
    "TenantA": {
      "IdentityOptions": {
        "Lockout": {
          "AllowedForNewUsers": true,
          "DefaultLockoutTimeSpan ": "00:05:00",
          "MaxFailedAccessAttempts ": 5
        },
        "Password": {
          "RequireDigit": false,
          "RequireLowercase": true,
          "RequireUppercase": true,
          "RequireNonAlphanumeric": false,
          "RequiredUniqueChars": 3,
          "RequiredLength": 6
        }
      }
    }
  }
}
```

**Reply by mariojsnunes** [reply]
Awesome! I will try!

**Reply by mariojsnunes** [reply]
I can confirm the `IConfigureOptions` implementation works :)
Thanks!

**Reply by xperiandri** [reply]
So is there no default functionality that will work out of the box like this?
``` JSON
  "OrchardCore": {
    "Default": {
      "SQL": {
        "ConnectionString": ""
      },
      "RawData": {
        "ConnectionString": ""
      },
      "EventStore": {
        "ConnectionString": ""
      }
    },
    "Sandbox": {
      "SQL": {
        "ConnectionString": ""
      },
      "RawData": {
        "ConnectionString": ""
      },
      "EventStore": {
        "ConnectionString": ""
      }
    },
```

**Reply by Piedone** [reply]
With `IShellConfiguration` such tenant-level configuration will work, yes.
